src/TodoList.tsx

```diff
@@ -0,0 +1,54 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Todo } from './react-app-env';
+import { TodoItem } from './TodoItem';
+
+type Props = {
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+};
+
+export const TodoList: React.FC<Props> = ({ todos, setTodos }) => {
+  const [editTodoId, setEditTodoId] = useState(0);
+  const { pathname } = useLocation();
+  let todosFilter = todos;
+
+  const filter = () => {
+    if (pathname === '/active') {
+      todosFilter = todos.filter(todo => !todo.completed);
+    }
+
+    if (pathname === '/completed') {
+      todosFilter = todos.filter(todo => todo.completed);
+    }
+
+    if (pathname === '/') {
+      todosFilter = todos;
+    }
+  };
+
+  filter();
```

Comment:
Consider moving the filter() call inside the useMemo or useEffect mentioned earlier to avoid unnecessary re-renders 🚀